scipy/signal/_filter_design.py

```diff
@@ -1633,6 +1654,19 @@ def zpk2sos(z, p, k, pairing=None, *, analog=False):
     # 4. Further optimizations of the section ordering / pole-zero pairing.
     # See the wiki for other potential issues.
 
+    xp = array_namespace(z, p)
+
+    # convert to numpy, convert back on exit   XXX
+    z, p = map(np.asarray, (z, p))
+
+    k = xp.asarray(k)
+    if xp.isdtype(k.dtype, 'complex floating'):
+        if xp.imag(k) != 0:
+            raise ValueError('k must be real')  
```

Comment:
I'm in two minds, would it be clearer to say zero imaginary part as the dtype doesn't have to be real?

Reply:
also can you add test coverage for this new check

Reply:
Actually, I'm no longer sure why this is here at all. Will double-check.